processing/src/main/java/io/druid/query/DruidMetrics.java

```diff
@@ -23,18 +23,47 @@
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Lists;
 import com.metamx.emitter.service.ServiceMetricEvent;
+import io.druid.query.aggregation.AggregatorFactory;
 import org.joda.time.Interval;
 
+import java.util.List;
+
 /**
  */
-public class QueryMetricUtil
+public class DruidMetrics
 {
-  public static <T> ServiceMetricEvent.Builder makeQueryTimeMetric(Query<T> query)
+  public final static String DATASOURCE = "dataSource";
+  public final static String TYPE = "type";
+  public final static String INTERVAL = "interval";
+  public final static String ID = "id";
+  public final static String STATUS = "status";
+
+  // task metrics
+  public final static String TASK_TYPE = "taskType";
+  public final static String TASK_STATUS = "taskStatus";
+
+  public final static String SERVER = "server";
+
+  public static int findNumComplexAggs(List<AggregatorFactory> aggs)
+  {
+    int retVal = 0;
+    for (AggregatorFactory agg : aggs) {
+      // This needs to change when we have support column types better
+      if (!agg.getTypeName().equals("float") && !agg.getTypeName().equals("long")) {
```

Comment:
HistogramAggregatorFactory throws UnsupportedOperationException on getTypeName(), 
need to check other aggregator factories too.